Description: Add shell completions back
 Upstream removed them in crates.io release. Content picked at release 1.6.1,
 commit 7c371a6852d4b2bdad280be766283abbf93e56ff.
 They discussed about auto generating shell completions, but hasn't seen
 progress: https://github.com/dbrgn/tealdeer/issues/220, also #256 #295
Last-Update: 2023-02-09
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- /dev/null
+++ b/completion/bash_tealdeer
@@ -0,0 +1,34 @@
+# tealdeer bash completion
+
+_tealdeer()
+{
+	local cur prev words cword
+	_init_completion || return
+
+	case $prev in
+		-h|--help|-v|--version|-l|--list|-u|--update|--no-auto-update|-c|--clear-cache|--pager|-r|--raw|--show-paths|--seed-config|-q|--quiet)
+			return
+			;;
+		-f|--render)
+			_filedir
+			return
+			;;
+		-p|--platform)
+			COMPREPLY=( $(compgen -W 'linux macos sunos windows android' -- "${cur}") )
+			return
+			;;
+		--color)
+			COMPREPLY=( $(compgen -W 'always auto never' -- "${cur}") )
+			return
+			;;
+	esac
+
+	if [[ $cur == -* ]]; then
+		COMPREPLY=( $( compgen -W '$( _parse_help "$1" )' -- "$cur" ) )
+		return
+	fi
+
+	COMPREPLY=( $(compgen -W '$( tldr -l | tr -d , )' -- "${cur}") )
+}
+
+complete -F _tealdeer tldr
--- /dev/null
+++ b/completion/fish_tealdeer
@@ -0,0 +1,26 @@
+#
+# Completions for the tealdeer implementation of tldr
+# https://github.com/dbrgn/tealdeer/
+#
+
+complete -c tldr -s h -l help           -d 'Print the help message.' -f
+complete -c tldr -s v -l version        -d 'Show version information.' -f
+complete -c tldr -s l -l list           -d 'List all commands in the cache.' -f
+complete -c tldr -s f -l render         -d 'Render a specific markdown file.' -r
+complete -c tldr -s p -l platform       -d 'Override the operating system.' -xa 'linux macos sunos windows android'
+complete -c tldr -s L -l language       -d 'Override the language' -x
+complete -c tldr -s u -l update         -d 'Update the local cache.' -f
+complete -c tldr      -l no-auto-update -d 'If auto update is configured, disable it for this run.' -f
+complete -c tldr -s c -l clear-cache    -d 'Clear the local cache.' -f
+complete -c tldr      -l pager          -d 'Use a pager to page output.' -f
+complete -c tldr -s r -l raw            -d 'Display the raw markdown instead of rendering it.' -f
+complete -c tldr -s q -l quiet          -d 'Suppress informational messages.' -f
+complete -c tldr      -l show-paths     -d 'Show file and directory paths used by tealdeer.' -f
+complete -c tldr      -l seed-config    -d 'Create a basic config.' -f
+complete -c tldr      -l color          -d 'Controls when to use color.' -xa 'always auto never'
+
+function __tealdeer_entries
+    tldr --list | string replace -a -i -r "\,\s" "\n"
+end
+
+complete -f -c tldr -a '(__tealdeer_entries)'
--- /dev/null
+++ b/completion/zsh_tealdeer
@@ -0,0 +1,47 @@
+#compdef tldr
+
+_applications() {
+    local -a commands
+    commands=(${(uonzf)"$(tldr --list 2>/dev/null)"//:/\\:})
+    _describe -t commands 'command' commands
+}
+
+_tealdeer() {
+    local I="-h --help -v --version"
+    integer ret=1
+    local -a args
+
+    args+=(
+        "($I -l --list)"{-l,--list}"[List all commands in the cache]"
+        "($I -f --render)"{-f,--render}"[Render a specific markdown file]:file:_files"
+        "($I -p --platform)"{-p,--platform}'[Override the operating system]:platform:((
+            linux
+            macos
+            sunos
+            windows
+            android
+        ))'
+        "($I -L --language)"{-L,--language}"[Override the language settings]:lang"
+        "($I -u --update)"{-u,--update}"[Update the local cache]"
+        "($I)--no-auto-update[If auto update is configured, disable it for this run]"
+        "($I -c --clear-cache)"{-c,--clear-cache}"[Clear the local cache]"
+        "($I)--pager[Use a pager to page output]"
+        "($I -r --raw)"{-r,--raw}"[Display the raw markdown instead of rendering it]"
+        "($I -q --quiet)"{-q,--quiet}"[Suppress informational messages]"
+        "($I)--show-paths[Show file and directory paths used by tealdeer]"
+        "($I)--seed-config[Create a basic config]"
+        "($I)--color[Controls when to use color]:when:((
+            always
+            auto
+            never
+        ))"
+        '(- *)'{-h,--help}'[Display help]'
+        '(- *)'{-v,--version}'[Show version information]'
+        '1: :_applications'
+    )
+
+    _arguments $args[@] && ret=0
+    return ret
+}
+
+_tealdeer
